### PR TITLE
WIP: pkgsEmscripten: emscripten package set using cross toolchain

### PR DESCRIPTION
**Pull request by FRidh**
Implement the emscripten package set using the cross toolchain in a similar
way as `pkgsMusl`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Unfortunately I'm getting an evaluation error. @dtzWill the structure is now roughly based on `pkgsMusl`.  Could you have a look at it? I suppose it's related to the splicing.

**Comment by matthewbauer** [comment]
Looks okay. TBH I'd like everybody to use WASM/WASI, but I suppose it's nice to work with emscripten as it's more popular. emconfigure and emmake are mostly just a collection of hacks, so the likelihood of working with many packages out of the box is not great.

File: pkgs/stdenv/emscripten-adapter.nix [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
 
-pkgs.stdenv.mkDerivation (
-  args // 
-  {
-
-  name = "emscripten-${args.name}";
-  buildInputs = [ emscripten python ] ++ buildInputs;
-  nativeBuildInputs = [ emscripten python ] ++ nativeBuildInputs;
+#   name = "emscripten-${args.name}";
+  buildInputs = (args.buildInputs or []) ++ [ emscripten python ];
```
Comment (matthewbauer) [inline pkgs/stdenv/emscripten-adapter.nix]: These won't cross compile

File: pkgs/top-level/emscripten-packages.nix [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+
+with super;
+
+{
+  stdenv = super.stdenvAdapters.emscriptenAdapter super.stdenv;
+
+  json_c = super.json_c.overrideDerivation (old: {
       nativeBuildInputs = [ autoreconfHook pkgconfig ];
       propagatedBuildInputs = [ zlib ];
       buildInputs = old.buildInputs ++ [ automake autoconf ];
```
Comment (matthewbauer) [inline pkgs/top-level/emscripten-packages.nix]: Same thing here.

**Comment by matthewbauer** [comment]
It would also be good to update emscripten. I believe it's a little bit out of date.

**Comment by FRidh** [comment]
> I'd like everybody to use WASM/WASI

I agree with you. Right now, I just want to see if we can get [pyodide](https://github.com/iodide-project/pyodide) running.

**Comment by FRidh** [comment]
I won't continue with this.

**Pull request closed without merging**